dashboard-ui/src/lib/color-shape.tsx

```diff
@@ -0,0 +1,224 @@
+    const colorSeed = view.getUint32(0);
+
+    let shapeIndex: number;
+    let colorIndex: number;
+
+    if (orderIndex < 20) {
+      // First 20 containers in order: All circles with unique colors
+      shapeIndex = 0; // Circle
+      colorIndex = orderIndex % 20;
+    } else if (orderIndex < 80) {
+      // Next 60 containers: Random shapes (excluding circle) with random colors
+      const nonCircleShapes = [1, 2, 3]; // Square, Diamond, Triangle
+      const shapeSeed = view.getUint32(4);
+      shapeIndex = nonCircleShapes[shapeSeed % nonCircleShapes.length];
+      colorIndex = colorSeed % 20;
+    } else {
+      // 80+ containers: Random shapes (all 4) with random colors
+      const shapeSeed = view.getUint32(4);
+      shapeIndex = shapeSeed % AVAILABLE_SHAPES.length;
+      colorIndex = colorSeed % 20;
+    }
+
+    return { shapeIndex, colorIndex };
+  };
+
+  const [shape, setShape] = useState<Shape>(Shape.CIRCLE);
+  const [color, setColor] = useState<string>('#000000');
+
+  useEffect(() => {
+    const colors = theme === 'dark' ? DARK_THEME_COLORS : LIGHT_THEME_COLORS;
+
+    getAssignment(containerKey).then(({ shapeIndex, colorIndex }) => {
+      setShape(AVAILABLE_SHAPES[shapeIndex]);
+      setColor(colors[colorIndex]);
+    });
+  }, [containerKey, theme]);
+
+  const sizeClasses = size === 'small' ? 'w-[8px] h-[8px]' : 'w-[13px] h-[13px]';
+
+  const baseStyle = {
+    backgroundColor: color,
+    display: 'inline-block',
+  };
+
+  switch (shape) {
+    case Shape.CIRCLE:
+      return (
+        <div
+          className={`${sizeClasses} rounded-full ${className}`}
+          style={baseStyle}
+        />
+      );
+    case Shape.SQUARE:
+      return (
+        <div
+          className={`${sizeClasses} ${className}`}
+          style={baseStyle}
+        />
+      );
+    case Shape.DIAMOND:
+      return (
+        <div
+          className={`${sizeClasses} ${className}`}
+          style={{
+            ...baseStyle,
+            transform: 'rotate(45deg)',
+          }}
+        />
+      );
+    case Shape.TRIANGLE:
+      return (
+        <div
+          className={`${sizeClasses} ${className}`}
+          style={{
+            ...baseStyle,
+            width: 0,
+            height: 0,
+            backgroundColor: 'transparent',
+            borderLeft: size === 'small' ? '4px solid transparent' : '6.5px solid transparent',
+            borderRight: size === 'small' ? '4px solid transparent' : '6.5px solid transparent',
+            borderBottom: size === 'small' ? `8px solid ${color}` : `13px solid ${color}`,
+          }}
+        />
+      );
+    default:
+      return (
+        <div
+          className={`${sizeClasses} rounded-full ${className}`}
+          style={baseStyle}
+        />
+      );
+  }
+};
+
+export { LIGHT_THEME_COLORS, DARK_THEME_COLORS };
+
+// Export utility function to clear container order (for testing/debugging)
+export const clearContainerOrder = (): void => {
```

Comment:
I don't think this is needed in the code, for debugging we can delete the storage using the browser tools